lib/shared/src/models/index.ts

```diff
@@ -1,70 +1,213 @@
+import { type AuthStatus, isCodyProUser, isEnterpriseUser, isFreeUser } from '../auth/types'
 import { fetchLocalOllamaModels } from '../llm-providers/ollama/utils'
 import { CHAT_INPUT_TOKEN_BUDGET, CHAT_OUTPUT_TOKEN_BUDGET } from '../token/constants'
-import type { ModelContextWindow, ModelUsage } from './types'
+import { ModelTag } from './tags'
+import { type ChatModel, type EditModel, type ModelContextWindow, ModelUsage } from './types'
 import { getModelInfo } from './utils'
 
+export type ModelId = string
+export type ApiVersionId = string
+export type ProviderId = string
+
+export type ModelRef = `${ProviderId}::${ApiVersionId}::${ModelId}`
+
+export type ModelCapability = 'chat' | 'autocomplete'
+export type ModelCategory = 'accuracy' | 'balanced' | 'speed'
+export type ModelStatus = 'experimental' | 'beta' | 'stable' | 'deprecated'
+export type ModelTier = 'free' | 'pro' | 'enterprise'
```

Comment:
```suggestion
export type ModelCategory = ModelTag.Accuracy | ModelTag.Balanced | ModelTag.Speed
export type ModelStatus = ModelTag.Experimental | ModelTag.Experimental | 'stable' | ModelTag.Deprecated
export type ModelTier = ModelTag.Free | ModelTag.Pro | ModelTag.Enterprise
```

safer / to avoid typos when the list gets longer?